pyrene/src/components/DateTimePicker/DateTimePicker.tsx

```diff
@@ -0,0 +1,290 @@
+  onBlur?: () => OnFunction,
+  onCalendarOpen?: () => void,
+  onClickOutside?(event: React.MouseEvent<HTMLDivElement>): void,
+  /**
+  * Function to handle onChange event
+  */
+  onChange: OnFunction,
+  /**
+   * This is a string array that represents the start and end labels of the component
+   */
+  range?: boolean,
+  /**
+   * Boolean to indicate if this input selects the end date of a range
+   */
+  selectEnd?: boolean,
+  /**
+  * Boolean to indicate if this input selects the start date of a range
+  */
+  selectStart?: boolean,
+  /**
+  * This is a Date object that represents the start date of the component
+  */
+  startDate?: Date,
+  tabNum?: number,
+  /**
+   * This is a unix timestamp, which is the number of seconds that have elapsed since Unix epoch
+   */
+  timeStamp?: number | null
+  /**
+   * This is must be a IANA time zone string
+   */
+  timeZone?: string,
+}
+
+const DateTimePicker: React.FC<DateTimePickerProps> = ({
+  calendarOpened,
+  closeOnSelect = true,
+  customCalendar,
+  dateOnly = false,
+  endDate,
+  inline = false,
+  label,
+  maxDateTime = getFutureDate({ years: 1 }),
+  minDateTime = 0,
+  name,
+  onCalendarOpen,
+  onChange,
+  onClickOutside,
+  range = false,
+  selectEnd,
+  selectStart,
+  startDate,
+  tabNum,
+  timeStamp,
+  timeZone = 'Europe/Zurich',
+}: DateTimePickerProps) => {
+
+  const [internalDate, setInternalDate] = useState<Date | undefined>();
+
+  const [dateValue, setDateValue] = useState('');
+  const [timeValue, setTimeValue] = useState('');
+
+  const [invalidTimestamp, setInvalidTimestamp] = useState(false);
+  const [errorValue, setErrorValue] = useState('');
+
+  // Sets internal date and passes validated value to parent
+  const handleOn = useCallback((dateString: string, timeString: string, onFunction?: OnFunction) => {
+    const date = getDateTypeFromddmmyyyyWithSep(dateString);
+    const time = getTimeTypeFromhhmmWithSep(timeString);
+
+    const validDateState = isValidDate(date);
+    const validTimeState = isValidTime(time);
+
+
+    if (dateOnly && date && validDateState) {
+      setInternalDate(convertToUTCtime(`${convertDateTypeToString(date)}`, timeZone));
+      if (onFunction) {
+        onFunction(convertToUTCtime(`${convertDateTypeToString(date)}`, timeZone).valueOf());
+      }
+    } else if (!dateOnly && date && time && validDateState && validTimeState) {
+      setInternalDate(convertToUTCtime(`${convertDateTypeToString(date)} ${convertTimeTypeToString(time)}`, timeZone));
+      if (onFunction) {
+        onFunction(convertToUTCtime(`${convertDateTypeToString(date)} ${convertTimeTypeToString(time)}`, timeZone).valueOf());
+      }
+    } else {
+      setInternalDate(undefined);
+
+      if (onFunction) {
+        onFunction(undefined);
+      }
+    }
+  }, [dateOnly, timeZone]);
+
+  // Handle changes from react datepicker
+  const onChangeReactDP = (date: Date | [Date, Date] | null, event: React.SyntheticEvent<any> | undefined): void => {
```

Comment:
`onChangeReactDP` is 50 lines of code long. Quite hard to follow,

Reply:
✅